tools/lisa-build-asset

```diff
@@ -60,22 +70,54 @@ def list_assets(recipe_dir):
 def get_asset_recipe(asset, recipe_dir):
     return os.path.join(recipe_dir, '{}.recipe'.format(asset))
 
-def make(asset, actions, arch, toolchain, recipe_dir, build_dir):
+def get_recipe_name(asset):
+    return asset + ".recipe"
+
+def make_native_build(asset, arch, recipe_dir, build_dir, alpine_source_dir, arch_chroot_dir, env):
+    # Install the chroot
+    subprocess.check_call(['sudo', os.path.join(alpine_source_dir, 'alpine-chroot-install'),
+            '-d', arch_chroot_dir, '-i', build_dir, '-a', QEMU_ARCH_NAMES[arch],
+            '-p', ' '.join(ALPINE_PKG),
+            '-k', ' '.join(env['LISA_ASSET_RECIPE_USED_ENV'].split())],
+            cwd=build_dir)
+
+    # Copy the recipe where the sources are and append the function to execute
+    shutil.copyfile(get_asset_recipe(asset, recipe_dir), os.path.join(build_dir, asset + ".recipe"))
+    with open(os.path.join(build_dir, get_recipe_name(asset)), "a") as f:
```

Comment:
That's neither robust nor necessary (it's not really supposed to happen but nothing prevents anyone from having an `exit(0)` in the middle of the script, and it prevents sourcing the script again to do the install stage for example).

It is probably possible to fuse that function with `make()` with a `chroot=None` parameter that is set to `arch_chroot_dir` when we want to use the Alpine chroot:
```
cmd  = ... # set by existent make() code

if chroot:
    cmd = [
        os.path.join(arch_chroot_dir, 'enter-chroot'),
        '-u', os.environ['USER'],
        '--', # If the script supports it, it's a very good idea to use -- to avoid conflicts between the custom command and the script options
        *cmd,
    ]

return subprocess.check_output(cmd, env=env, cwd=build_dir)
```

Reply:
I agree this is not really gracious. I did this as a workaround not to modify the `enter-chroot` script. I will make a PR to the alpine-install-chroot repository once I have the right to.